### DISCUSSION TITLE
Version 4 and 5

### DISCUSSION THREAD

**Discussion by rejas** [Modernizr/Modernizr#2638] [Ideas]
The current official branch, v3, only gets bugfixes. For quite some time the (main) v4 branch lingers around with a commit here and there, with the main work (es6) not being ready.

In order to freshen it up and get some momentum again I propose the following:
- we work towards a v4 branch that contains stuff like the removal/rearranging of tests, basically all that is in the main branch wihtout the es6. 
- the es6 stuff should be worked on again but it would be released when its ready in a v5 

Oh, and anyone against changing the name of the master branch to main?

Would lvoe to hear from you regulars @Markel  @patrickkettner  @roblarsen  :-)

**Comment by Markel** [comment]
> Note: When I talk about v4 in this message I'm referring to the v4 concept that you described, not the es6 module version 

First of all, about renaming to main, absolutely yes! Will we have to deal with git annoyances because they don't expect you to change branches names? Probably yes, but go for it :)

About the v4/v5, if I'm understanding this correctly the v4 release would be more or less what we have at the master/main branch right now, wouldn't it? Like there are a lot of PRs that had been merged but not released yet. I think that there's not any es6 stuff in the master branch right now. Correct me if I'm wrong. (I've seen #2639, but after writing this paragraph)

I'm guessing btw that the v4 release would not be updated in the website as the Netlify functions are in the es6 PR. About v5, if I'm really honest I'm kind of lost with it right now in my mind, it's been a long time since I invested a reasonable amount of time at #2594 (and the recent time has been between exams and new semester so I haven't retained a lot of information), but I feel that not too much need to be done, and most of it is rewriting as all feature detects are converted to es6 (even the ones that haven't been released yet, I've taken care of it) and the tests work correctly.

Also, remember that #2594 can be merged into a _development_ branch of some sorts, my biggest objective has been maintaining it up to date.

But yeah, summarizing (I write way too much) I don't see it as bad idea what you've proposed :)
